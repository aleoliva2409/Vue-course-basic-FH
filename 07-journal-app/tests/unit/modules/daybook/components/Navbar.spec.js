import { shallowMount } from '@vue/test-utils';

import NavBar from '@/modules/daybook/components/NavBar.vue';
import createVuexStore from '../../../mocks/mock-store';

// ?? hacer la configuración individual en el archivo si se trabaja con options api y composition api porque si se hace esta configuración global rompe el router del options api
// import {
//   VueRouterMock,
//   createRouterMock,
//   injectRouterMock,
// } from 'vue-router-mock';
// import { config } from '@vue/test-utils';

// const router = createRouterMock();
// beforeEach(() => {
//   injectRouterMock(router);
// });

// config.plugins.VueWrapper.install(VueRouterMock);

describe('Tests Navbar component', () => {
  const store = createVuexStore({
    user: {
      name: 'Alejandro Oliva',
      email: 'ale@gmail.com',
    },
    status: 'authenticated',
    idToken: 'ABC',
    refreshToken: 'XYZ',
  });

  beforeEach(() => jest.clearAllMocks());

  test('should show component', () => {
    const wrapper = shallowMount(NavBar, {
      global: {
        plugins: [store],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should redirect and logout when button is clicked', async () => {
    const wrapper = shallowMount(NavBar, {
      global: {
        plugins: [store],
      },
    });

    await wrapper.find('button').trigger('click');

    expect(wrapper.router.push).toHaveBeenCalledWith({ name: 'login' });

    expect(store.state.auth).toEqual({
      user: null,
      status: 'not-authenticated',
      idToken: null,
      refreshToken: null,
    });
  });
});
