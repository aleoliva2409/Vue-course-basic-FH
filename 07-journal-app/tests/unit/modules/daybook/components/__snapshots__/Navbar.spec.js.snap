// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Navbar component should show component 1`] = `
<nav class="navbar bg-primary"><a class="navbar-brand text-white" href="#"><img src="" alt="logo" height="24" class="d-inline-block align-text-top mx-2"> Alejandro Oliva</a>
  <div class="d-flex"><button class="btn btn-outline-info mx-2"><i class="fa fa-sign-out-alt"></i></button></div>
</nav>
`;
